packedForest/src/forestTypes/unsupervisedForests/urerf/urerfTree.h

```diff
@@ -0,0 +1,267 @@
+#ifndef urerfTree_h
+#define urerfTree_h
+#include "../../../baseFunctions/fpBaseNode.h"
+#include "unprocessedURerFNode.h"
+#include <vector>
+#include <map>
+#include <assert.h>
+
+namespace fp{
+
+	template <typename T>
+		class urerfTree
+		{
+			protected:
+				float OOBAccuracy;
+				float correctOOB;
+				float totalOOB;
+				std::vector< fpBaseNode<T, std::vector<int> > > tree;
+				std::vector< unprocessedURerFNode<T> > nodeQueue;
+				std::vector< unprocessedURerFNode<T> > leafNodes;
+
+			public:
+				urerfTree() : OOBAccuracy(-1.0),correctOOB(0),totalOOB(0){}
+
+				void loadFirstNode(){
+					nodeQueue.emplace_back(fpSingleton::getSingleton().returnNumObservations());
+				}
+
+				inline bool shouldProcessNode(){
+					if(nodeQueue.back().returnNodeImpurity()<=0){
+						return false;
+					}
+					if(nodeQueue.back().returnInSampleSize() <= fpSingleton::getSingleton().returnMinParent()){
+						return false;
+					}
+                if(nodeQueue.back().returnDepth() >= fpSingleton::getSingleton().returnMaxDepth()){
+                          return false;
+                      }
+					return true;
+				}
+
+				inline float returnOOB(){
+					return correctOOB/totalOOB;
+				}
+
+				inline int returnLastNodeID(){
+					return tree.size()-1;
+				}
+
+				inline void linkParentToChild(){
+					if(nodeQueue.back().returnIsLeftNode()){
+						tree[nodeQueue.back().returnParentID()].setLeftValue(returnLastNodeID());
+					}else{
+						tree[nodeQueue.back().returnParentID()].setRightValue(returnLastNodeID());
+					}
+				}
+
+
+				inline int returnMaxDepth(){
+					int maxDepth=0;
+					for(auto nodes : tree){
+						if(maxDepth < nodes.returnDepth()){
+							maxDepth = nodes.returnDepth();
+						}
+					}
+					return maxDepth;
+				}
+
+
+				inline int returnNumLeafNodes(){
+					int numLeafNodes=0;
+					for(auto nodes : tree){
+						if(!nodes.isInternalNode()){
+							++numLeafNodes;
+						}
+					}
+					return numLeafNodes;
+				}
+
+
+
+                                inline void updateSimMat(std::map<int, std::map<int, int> > &simMat, std::map<std::pair<int, int>, double> &pairMat){
```

Comment:
some weird formatting in here

Reply:
Fixed in 5f65da04650742ed24365cde194085e56b01a60d